### PR TITLE
Configure smartstack provides custom labels for services

### PR DESCRIPTION
**Pull request by mcarlson-zero**
like $SERVICENAME$SERVCEPORT*, loads them as yaml, and applies them as
labels to that service on that port.

### CODE REVIEW

**Comment by mcarlson-zero** [comment]
@solarkennedy @jolynch bump?

File: src/nerve_tools/configure_nerve.py [hunk 1]
```diff
@@ -75,6 +79,14 @@ def generate_subconfiguration(
     # hacheck will simply ignore the healthcheck_uri for TCP mode checks
     healthcheck_uri = service_info.get('healthcheck_uri', '/status')
     healthcheck_mode = service_info.get('healthcheck_mode', mode)
+    custom_labels = {}
+    try:
+        path = os.path.join(labels_dir, service_name + str(port) + '*')
+        for label_file in glob(path):
+            with open(label_file) as f:
+                custom_labels.update(yaml.load(f))
```
Comment (solarkennedy) [inline src/nerve_tools/configure_nerve.py]: safe_load

File: src/tests/configure_nerve_test.py [hunk 2]
```diff
@@ -125,10 +128,15 @@ def test_generate_subconfiguration():
...
             'labels': {
+                'label1': 'value1',
+                'label2': 'value2',
                 'region:another_region': '',
             },
         },
     }
+    labels_dir = tmpdir.mkdir('itest_configure_nerve_labels')
+    with open(labels_dir.join('test_service1234').strpath, 'w') as f:
+        f.write(yaml.dump({'label1': 'value1', 'label2': 'value2'}))
```
Comment (solarkennedy) [inline src/tests/configure_nerve_test.py]: do we have to do fileio in the unit test?

**Comment by mcarlson-zero** [comment]
@solarkennedy We want to use Nerve to register Cascade masters, and put the application version in the labels so we can read that from Zookeeper. CASCADE-57 has more information on the motivation and a link to the RFC.

File: src/nerve_tools/configure_nerve.py (hunk 1 above)
Comment (solarkennedy) [inline src/nerve_tools/configure_nerve.py]: This is how you can use safe_load with the CLoader
https://github.com/Yelp/service_configuration_lib/blob/master/service_configuration_lib/__init__.py#L26

**Comment by solarkennedy** [comment]
@mcarlson-zero ready to ship. Would you like me to press the button?

**Pull request merged**